site/lib/moralis.ts

```diff
@@ -32,3 +32,35 @@ export const getPledgeByAddress = async (address: string) => {
 
   return await query.equalTo("address", address).first();
 };
+
+export const findOrCreatePledge = async (params) => {
+  await MoralisClient;
+
+  const userSession = await findUserSession(params.sessionToken);
+  if (!userSession) throw new Error("Invalid Session");
+
+  const user = userSession?.get("user");
+  const acl = new Moralis.ACL();
+  acl.setPublicReadAccess(true);
+  acl.setWriteAccess(user.id, true);
+
+  const Pledge = Moralis.Object.extend("Pledge");
+  const newPledge = params.pledge.objectId
+    ? await new Moralis.Query(Pledge).get(params.pledge.objectId)
```

Comment:
Would be nice to skip this `get` roundtrip and just set directly.... Buuuut we need to handle the case I outlined in the comment below: